px4_ros2_cpp/src/navigation/experimental/local_navigation_interface.cpp

```diff
@@ -0,0 +1,281 @@
+/****************************************************************************
+ * Copyright (c) 2023 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+
+#include <rclcpp/rclcpp.hpp>
+#include <px4_ros2/navigation/experimental/local_navigation_interface.hpp>
+
+namespace px4_ros2
+{
+
+LocalNavigationInterface::LocalNavigationInterface(
+  rclcpp::Node & node, const uint8_t pose_frame,
+  const uint8_t velocity_frame)
+: NavigationInterfaceBase(node)
+{
+  // Check that selected pose and velocity reference frames are valid
+  const auto * it_pose_frame = std::find(
+    std::begin(kAvailablePoseFrames), std::end(
+      kAvailablePoseFrames), pose_frame);
+  if (it_pose_frame == std::end(kAvailablePoseFrames)) {
+    RCLCPP_WARN(
+      node.get_logger(), "Failed to initialize LocalNavigationInterface: invalid pose reference frame %i. Setting pose reference frame to POSE_FRAME_UNKNOWN.",
+      pose_frame);
+    _pose_frame = AuxLocalPosition::POSE_FRAME_UNKNOWN;
+  } else {
+    _pose_frame = pose_frame;
+  }
+
+  const auto * it_vel_frame = std::find(
+    std::begin(kAvailableVelocityFrames), std::end(
+      kAvailableVelocityFrames), velocity_frame);
+  if (it_vel_frame == std::end(kAvailableVelocityFrames)) {
+    RCLCPP_WARN(
+      node.get_logger(), "Failed to initialize LocalNavigationInterface: invalid velocity reference frame %i. Setting velocity reference frame to VELOCITY_FRAME_UNKNOWN.",
+      velocity_frame);
+    _velocity_frame = AuxLocalPosition::VELOCITY_FRAME_UNKNOWN;
+  } else {
+    _velocity_frame = velocity_frame;
+  }
```

Comment:
Defining a separate enum would make these a bit nicer to use (instead of uint8_t), with the small downside of having to map from the enum to `AuxLocalPosition::POSE_FRAME_x`.